Mathlib/Analysis/Calculus/ContDiff/FaaDiBruno.lean

```diff
@@ -786,24 +797,21 @@ variable (𝕜 E F G) in
 /-- Bundled version of `compAlongOrderedFinpartition`, depending continuously linearly on `f`
 and continuously multilinearly on `p`.-/
 noncomputable def compAlongOrderedFinpartitionL :
-    (ContinuousMultilinearMap 𝕜 (fun _i : Fin c.length ↦ F) G) →L[𝕜]
-      ContinuousMultilinearMap 𝕜 (fun i : Fin c.length ↦ (E[×c.partSize i]→L[𝕜] F))
-        (E[×n]→L[𝕜] G) := by
-  refine MultilinearMap.mkContinuousLinear c.compAlongOrderedFinpartitionₗ 1 (fun f p ↦ ?_)
-  simp only [one_mul]
-  change ‖c.compAlongOrderedFinpartition f p‖ ≤ _
-  apply ContinuousMultilinearMap.opNorm_le_bound (by positivity) (fun v ↦ ?_)
-  simp only [compAlongOrderFinpartition_apply]
-  apply (f.le_opNorm _).trans
-  rw [mul_assoc, ← c.prod_sigma_eq_prod, ← Finset.prod_mul_distrib]
-  gcongr with m _
-  exact (p m).le_opNorm _
-
-@[simp] lemma compAlongOrderedFinpartitionL_apply
-    (f : ContinuousMultilinearMap 𝕜 (fun _i : Fin c.length ↦ F) G)
+    (F [×c.length]→L[𝕜] G) →L[𝕜]
+      ContinuousMultilinearMap 𝕜 (fun i ↦ E[×c.partSize i]→L[𝕜] F) (E[×n]→L[𝕜] G) := by
+  refine MultilinearMap.mkContinuousLinear c.compAlongOrderedFinpartitionₗ 1 fun f p ↦ ?_
+  simp only [one_mul, compAlongOrderedFinpartitionₗ_apply_apply]
+  apply norm_compAlongOrderedFinpartition_le
+
+@[simp] lemma compAlongOrderedFinpartitionL_apply (f : F [×c.length]→L[𝕜] G)
     (p : ∀ (i : Fin c.length), E[×c.partSize i]→L[𝕜] F) :
     c.compAlongOrderedFinpartitionL 𝕜 E F G f p = c.compAlongOrderedFinpartition f p := rfl
 
+theorem norm_compAlongOrderedFinpartitionL_le :
+    set_option maxSynthPendingDepth 2 in
```

Comment:
@kim-em How do I debug why this is needed?

Reply:
At some point we suggested that it would probably be a good idea to have this as the default, but I don't remember what happened next.